tests/python/pants_test/backend/python/rules/test_python_fmt_integration.py

```diff
@@ -109,3 +123,51 @@ def test_black_should_format_python_code(self):
       self.assertEqual(CONSISTENTLY_FORMATTED_HELLO, after_formatting)
     self.assertIn("1 file reformatted", pants_run.stderr_data)
     self.assertNotIn("unchanged", pants_run.stderr_data)
+
+  def test_black_lint_should_fail_but_not_format_python_code(self):
+    with build_directory() as root_dir:
+      code = write_inconsistently_formatted_file(root_dir, "hello.py")
+      command = [
+        'lint-v2',
+        f'{root_dir}'
+      ]
+      pants_run = self.run_pants(command=command)
+      self.assert_failure(pants_run)
+      after_formatting = code.read_text()
+      self.assertEqual(INCONSISTENTLY_FORMATTED_HELLO, after_formatting)
+    self.assertIn("1 file would be reformatted", pants_run.stderr_data)
+    self.assertNotIn("1 file reformatted", pants_run.stderr_data)
+    self.assertNotIn("unchanged", pants_run.stderr_data)
+
+  def test_black_lint_given_multiple_files(self):
+    with build_directory() as root_dir:
+      write_inconsistently_formatted_file(root_dir, "incorrect.py")
+      write_inconsistently_formatted_file(root_dir, "broken.py")
+      write_consistently_formatted_file(root_dir, "pristine.py")
+      command = [
+        'lint-v2',
+        f'{root_dir}'
+      ]
+      pants_run = self.run_pants(command=command)
+      self.assert_failure(pants_run)
+    self.assertIn("2 files would be reformatted", pants_run.stderr_data)
```

Comment:
If we end up adding much more testing to this file, we could consider a helper method which accepts kwargs e.g. `files_reformatted=0, files_left_unchanged=0` to do some of this automatically.

Reply:
We could, and I'll keep it in mind in case more tests are added, but I kind of like the absence of indirection when reading unit tests, even to the cost of some duplication. Not a hard rule, though.